### PR TITLE
Update for kubernetes-client-api.version v6.8.1

### PR DESCRIPTION
**Pull request by julieheard**
Downstream of https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/221

1) When the above PR for kubernetes-client-api-plugin is merged and we get a new release, we need to check the new version number against what I have put in the pom for the version number on this plugin PR.
2) I have included a dependency on snakeyaml-api 2.0 which is currently not released - PR is here https://github.com/jenkinsci/snakeyaml-api-plugin/pull/94. When the new release version for kubernetes-client-api-plugin is sorted, we can re-run tests and check if the snakeyaml 2.0+ dependency is needed. If it is needed, we will also need a release of snakeyaml-api and will need to include that released version in this PR.

I will keep this PR in draft until the PR for kubernetes-client-api is merged and we can check the above points.

### Testing done
I ran mvn verify to test locally and all working, all tests passing

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.50</version>
+    <version>4.66</version>
```
Comment (julieheard) [inline pom.xml:6]: I bumped this to 4.66 due to needing a higher version of [configuration-as-code-plugin](https://github.com/jenkinsci/configuration-as-code-plugin).
Version [1647.ve39ca_b_829b_42](https://github.com/jenkinsci/configuration-as-code-plugin/releases/tag/1647.ve39ca_b_829b_42) onwards of configuration-as-code includes changes made for the [html 3.x update](https://github.com/jenkinsci/configuration-as-code-plugin/pull/2283). As you can see further down, I have bumped configuration-as-code to the latest release.

File: pom.xml [hunk 2]
```diff
@@ -40,15 +40,15 @@
     <changelist>-SNAPSHOT</changelist>
 
     <!-- jenkins versions -->
-    <jenkins.version>2.332.4</jenkins.version>
-    <bom.artifactId>bom-2.332.x</bom.artifactId>
-    <bom.version>1763.v092b_8980a_f5e</bom.version>
+    <jenkins.version>2.401.1</jenkins.version>
+    <bom.artifactId>bom-2.401.x</bom.artifactId>
+    <bom.version>2244.vd60654536b_96</bom.version>
```
Comment (julieheard) [inline pom.xml:45]: When initially bumping kubernetes-client-api, it needed a dependency on snakeyaml-api 2.0 and Jenkins needed to be at least 2.361.4 or higher.

File: pom.xml [hunk 3]
```diff
@@ -40,15 +40,15 @@
...
 
     <!-- dependency versions -->
     <authentication-tokens.version>1.3</authentication-tokens.version>
     <docker-commons.version>1.15</docker-commons.version>
     <google-oauth-plugin.version>0.8</google-oauth-plugin.version>
-    <kubernetes-client-api.version>6.3.1-206.v76d3b_6b_14db_b</kubernetes-client-api.version>
+    <kubernetes-client-api.version>6.8.1-220.v1f66736f9a_42</kubernetes-client-api.version>
```
Comment (julieheard) [inline pom.xml:51]: This version number will need to be checked when this PR is merged https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/221 and we have a new release version for kubernetes-client-api. It needs to be at least 6.8.0 to be compatible with the snakeyaml 2.0 changes.

File: src/main/java/org/jenkinsci/plugins/kubernetes/auth/impl/KubernetesAuthKubeconfig.java [hunk 4]
```diff
@@ -29,12 +29,8 @@ public ConfigBuilder decorate(ConfigBuilder builder, KubernetesAuthConfig config
...
 
     public io.fabric8.kubernetes.api.model.ConfigBuilder buildConfigBuilder(KubernetesAuthConfig config, String context, String clusterName, String username) throws KubernetesAuthException {
-        try {
-            io.fabric8.kubernetes.api.model.Config kubeConfig = KubeConfigUtils.parseConfigFromString(getKubeconfig());
-            return new io.fabric8.kubernetes.api.model.ConfigBuilder(kubeConfig);
-        } catch (IOException e) {
-            throw new KubernetesAuthException(e.getMessage(), e);
-        }
+        io.fabric8.kubernetes.api.model.Config kubeConfig = KubeConfigUtils.parseConfigFromString(getKubeconfig());
+        return new io.fabric8.kubernetes.api.model.ConfigBuilder(kubeConfig);
```
[the remark this answers is not in the document]
Comment (rsandell) [reply src/main/java/org/jenkinsci/plugins/kubernetes/auth/impl/KubernetesAuthKubeconfig.java:33]: You could change it to catch a general `Exception` perhaps?

**Comment by Vlatombe** [comment]
Hi @maxlaverse, we are looking into a coordinated releases of the plugins depending on https://github.com/jenkinsci/kubernetes-client-api-plugin (as this project tend to introduce breaking changes regularly). Could you give some indication of your availability to let this happen? Otherwise, would you be okay for a co-maintainership on this plugin?

File: pom.xml [hunk 5]
```diff
@@ -66,6 +66,11 @@
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>jackson2-api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>snakeyaml-api</artifactId>
+      <version>2.1-110.v4960f4074978</version>
```
Comment (julieheard) [inline pom.xml:72]: This is the incremental version of snakeyaml-api, needs to be updated when https://github.com/jenkinsci/snakeyaml-api-plugin/pull/94 is merged and released

**Comment by rsandell** [comment]
Looks like the pom needs to be modernised as well? The build error indicates that maven is running with an old jdk?

**Comment by maxlaverse** [comment]
@julieheard , could you merge master into your branch ? I've removed support for Java 8.

> Could you give some indication of your availability to let this happen?

I should be able to do the release within 1-2hours lag behind the merge of the downstream PR, on CEST day @Vlatombe. I'm definitely open to co-maintainership for this plugin.

**Comment by julieheard** [comment]
Done, thank you [Maxime](https://github.com/maxlaverse) 🙂

File: pom.xml (hunk 3 above)
Comment (Vlatombe) [reply pom.xml:51]: https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/221

File: pom.xml (hunk 5 above)
Comment (Vlatombe) [reply pom.xml:72]: https://github.com/jenkinsci/snakeyaml-api-plugin/pull/94

File: pom.xml (hunk 3 above)
Comment (Vlatombe) [inline pom.xml:51]:
```suggestion
    <kubernetes-client-api.version>6.8.1-224.vd388fca_4db_3b_</kubernetes-client-api.version>
```

**Comment by maxlaverse** [comment]
I've prepared my local environment for the release but I'm entering a car for ~6hours. I added you as collaborator in case I don't manage to release during a break and you want to have it done earlier.

**Comment by Vlatombe** [comment]
Thanks! I just checked that I already have deploy rights to repo.jenkins-ci.org on this plugin, so this should be fine.

**Pull request merged**